data/transactions/logic/evalStateful_test.go

```diff
@@ -169,6 +169,10 @@ func (l *testLedger) LatestTimestamp() int64 {
 	return int64(rand.Uint32() + 1)
 }
 
+func (l *testLedger) GetBlockTimeStamp(r basics.Round) (int64, error) {
```

Comment:
Returning a random stamp seems hard to test. Maybe we should just return something like 4.5s * round + some start epoch.

That way we get monotone increases and testability.